.github/workflows/ci-cd.yml

```
@@ -43,8 +43,21 @@ jobs:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
       
+      - name: Build versioned docker image
+        run: docker build --target production --tag ${{ secrets.DOCKER_USERNAME }}/todo-app:$GITHUB_SHA .
+
       - name: Build prod docker image
         run: docker build --target production --tag ${{ secrets.DOCKER_USERNAME }}/todo-app:prod .
```

If we preferred, we can combine these:
```suggestion
      - name: Build prod docker image
        run: docker build --target production --tag ${{ secrets.DOCKER_USERNAME }}/todo-app:prod --tag ${{ secrets.DOCKER_USERNAME }}/todo-app:$GITHUB_SHA .
```